### DISCUSSION TITLE
How to horizontally scale tasks run using KubernetesPodOperator using HPA?

### DISCUSSION THREAD

**Discussion by gowthamgutha** [apache/airflow#20065] [Q&A]
**Context**
I have a DAG which consists of tasks that use `KubernetesPodOperator`. 

**Requirement**
When these tasks reach a certain CPU/Memory limit, I would like another instance of this task to be run so that they share the load by scaling horizontally.

**Hypothesis**
Since, `KubernetesPodOperator` abstracts `Pod` rather than a `Deployment`, it doesn't fit into the [HPA definition](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/).

For operators, like `PythonOperator`, `BashOperator`, the worker pod itself does the task and we can scale the workers through HPA (if I am not wrong).
For `KubernetesPodOperator`, the worker pod merely starts the task pod defined in the DAG as `KubernetesPodOperator` and then goes to completed state once the task pod is started.

It is this task pod that I would like to scale horizontally based on CPU/Memory or other metrics. Is it possible?

**Comment by raphaelauv** [comment]
KubernetesPodOperator start a pod not a deployment , HPA is about scaling deployment

if your K8S do not scale compare to the number of KubernetesPodOperator you want run , then increase the size of the K8S nodepools where run your pods trigger by the KubernetesPodOperator
